src/sql/workbench/contrib/notebook/browser/cellViews/code.component.ts

```diff
@@ -146,8 +146,8 @@ export class CodeComponent extends AngularDisposable implements OnInit, OnChange
 			let connectionService = this.connectionService;
 			if (!shouldConnect && connectionService && connectionService.isConnected(cellUri)) {
 				connectionService.disconnect(cellUri).catch(e => this.logService.error(e));
-			} else if (shouldConnect && this._model.activeConnection && this._model.activeConnection.id !== '-1') {
-				connectionService.connect(this._model.activeConnection, cellUri).catch(e => this.logService.error(e));
+			} else if (shouldConnect && this._model.context && this._model.context.id !== '-1') {
```

Comment:
It would be good to have this -1 ID be defined as a const somewhere so it's clear that this is a special ID used for the default context.